### DISCUSSION TITLE
[Question][Hw5] order, join of PowSet

### DISCUSSION THREAD

**Discussion by HLe4s-kaist** [prosyslab-classroom/cs524-program-analysis#306] [Q&A] [answered]
Name: ChanU Kang

I am going through the codes given by hw5. but in moduel PowSet, there are some functions, order, join which I should implemented and they doens't have any parameter. 

so, I expected, 'let order p1 p2 = failwith "Not implemented"' but It is 'let order = failwith "Not implemented"'
and I'm not allowed to change the function type. except let to let rec. 

So this code involve that I must implement join or order of PowSet without any parameter.
But I can't imagine the way to compare without anything to compare or merge without anything to merge. 

Is this intented definition? Is there a way to do this end?

or...

Can I change it? because the actual type of order is 'val order : t -> t -> bool', as in CPO signature which is implemented by the PosSet?

**Comment by goodtaeeun** [comment, marked as answer]
As you can see in the case of `PowSet.bottom`, `PowSet.narrow`, and `PowSet.widen`,
they are defined with the functions from the Set module of Ocaml.
You can do the same with `order` and `join`.
